src/core/public/application/application_service.test.tsx

```diff
@@ -17,57 +17,219 @@
+    const startContract = await service.start({ http, injectedMetadata });
+
     expect(startContract.availableApps).toMatchInlineSnapshot(`
-Array [
-  Object {
-    "id": "app1",
-  },
-]
-`);
+                        Map {
+                          "app1" => Object {
+                            "id": "app1",
+                          },
+                        }
+                `);
     expect(startContract.availableLegacyApps).toMatchInlineSnapshot(`
-Array [
-  Object {
-    "id": "app2",
-  },
-]
-`);
+                        Map {
+                          "app2" => Object {
+                            "id": "app2",
+                          },
+                        }
+                `);
   });
 
   it('passes registered applications to capabilities', async () => {
     const service = new ApplicationService();
-    const setup = service.setup();
-    setup.registerApp({ id: 'app1' } as any);
+    const context = contextServiceMock.createSetupContract();
+    const setup = service.setup({ context });
+    setup.register(Symbol(), { id: 'app1' } as any);
+
+    const http = httpServiceMock.createStartContract();
     const injectedMetadata = injectedMetadataServiceMock.createStartContract();
-    await service.start({ injectedMetadata });
+    await service.start({ http, injectedMetadata });
+
     expect(MockCapabilitiesService.start).toHaveBeenCalledWith({
-      apps: [{ id: 'app1' }],
-      legacyApps: [],
+      apps: new Map([['app1', { id: 'app1' }]]),
+      legacyApps: new Map(),
       injectedMetadata,
     });
   });
 
   it('passes registered legacy applications to capabilities', async () => {
     const service = new ApplicationService();
-    const setup = service.setup();
+    const context = contextServiceMock.createSetupContract();
+    const setup = service.setup({ context });
     setup.registerLegacyApp({ id: 'legacyApp1' } as any);
+
+    const http = httpServiceMock.createStartContract();
     const injectedMetadata = injectedMetadataServiceMock.createStartContract();
-    await service.start({ injectedMetadata });
+    await service.start({ http, injectedMetadata });
+
     expect(MockCapabilitiesService.start).toHaveBeenCalledWith({
-      apps: [],
-      legacyApps: [{ id: 'legacyApp1' }],
+      apps: new Map(),
+      legacyApps: new Map([['legacyApp1', { id: 'legacyApp1' }]]),
       injectedMetadata,
     });
   });
+
+  it('returns renderable JSX tree', async () => {
+    const service = new ApplicationService();
+    const context = contextServiceMock.createSetupContract();
+    service.setup({ context });
+
+    const http = httpServiceMock.createStartContract();
+    const injectedMetadata = injectedMetadataServiceMock.createStartContract();
+    injectedMetadata.getLegacyMode.mockReturnValue(false);
+    const start = await service.start({ http, injectedMetadata });
+
+    expect(shallow(React.createElement(() => start.getComponent()))).toMatchSnapshot();
```

Comment:
Is there a way to check a method returns a JSX tree, without testing a tree content? The current test may lead to false-positive fails.